### ISSUE TITLE
Cannot generate an identity on Alpine Linux

### ISSUE THREAD

**Issue by djmoch** [str4d/age-plugin-yubikey#84]
## Environment

* OS: Alpine Linux 3.16.2
* age-plugin-yubikey version: 0.3.0

## What were you trying to do

Generate an identity to install to my Yubikey 4.

## What happened

```sh
$ ./age-plugin-yubikey
✨ Let's get your YubiKey set up for age! ✨

This tool can create a new age identity in a free slot of your YubiKey.
It will generate an identity file that you can use with an age client,
along with the corresponding recipient. You can also do this directly
with:
    age-plugin-yubikey --generate

If you are already using a YubiKey with age, you can select an existing
slot to recreate its corresponding identity file and recipient.

When asked below to select an option, use the up/down arrow keys to
make your choice, or press [Esc] or [q] to quit.

thread 'main' panicked at 'range end index 4 out of range for slice of length 0', /home/djmoch/.cargo/registry/src/github.com-1ecc6299db9ec823/yubikey-0.5.0/src/transaction.rs:160:9
```

**Comment by djmoch** [comment]
Without looking it further, I'm assuming this is an issue in the yubikey crate. I'll leave this here for feedback or until I can confirm.

**Comment by djmoch** [comment]
It turns out this was _not_ an issue with anything in the application, but with the Yubikey device itself. Are Yubikey 4 devices really supported? Here's the info on the non-functioning device.

```sh
$ ykman info
Device type: YubiKey 4
Firmware version: 4.3.3
Enabled USB interfaces: OTP, FIDO, CCID

Applications
FIDO2       	Not available	
OTP         	Enabled      	
FIDO U2F    	Enabled      	
OATH        	Enabled      	
YubiHSM Auth	Not available	
OpenPGP     	Enabled      	
PIV         	Enabled      	
```

Anyway, I ordered a Yubikey 5 and it appears to be working just fine.

**Comment by mihaigalos** [comment]
@djmoch, there's an Alpine Docker PR in case you want to use it: https://github.com/str4d/age-plugin-yubikey/pull/69.

**Comment by str4d** [comment]
This indeed looks like a problem with the `yubikey` crate. Supposedly we support YubiKey 4 there; this must be an edge case we haven't seen before. I'm curious as to whether it is a problem with that specific YubiKey, or with the YubiKey 4.3.3 firmware.

**Comment by djmoch** [comment]
@str4d I see it works with a Yubikey 4C FIPS with firmware 4.4.5. This is conversation is probably better handled over at the yubikey repo, so I'll open an issue there.

**Comment by str4d** [comment]
This issue was fixed by https://github.com/iqlusioninc/yubikey.rs/pull/466 which we brought into `main` in https://github.com/str4d/age-plugin-yubikey/commit/9418921dab67f7ba01f6ffce2a10f5e3ca1c4557 (fix confirmed by https://github.com/str4d/age-plugin-yubikey/issues/114#issuecomment-1407330664). The fix will be in version 0.4.0 (as it requires a new version of the `yubikey` crate).
